### PR TITLE
feat(article): Add native articles support

### PR DESCRIPTION
**Pull request by damassi**
### Description

> ⚠️ Behind `AREnableNativeArticleView` feature flag

This adds a standard and feature articles to the app, guards against other types of articles (and routes them into a webview) and adds the ability to share. 

Additionally, adds a new `<HTML />` component, used like so:

```tsx
<HTML 
  variant='lg' 
  tagsStyles={{ ...onlyOverrideIfYouMustUseVariant.... }}
/>
```

https://github.com/artsy/eigen/assets/236943/4d75e6dc-b8fa-45cc-8f6f-09654fa38fcd

Screenshot 2023-06-14 at 9 46 34 PM

cc @artsy/mobile-platform 

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Review by gkartalis — COMMENTED** [review]
the `RenderHtml` api looks really straightforward 🔥 exited about this feature! 

(I know that the PR is not in a finalised state yet, but noticed that when running locally I get this error after enabling the ff:




Is it a known issue or did I miss something setting up?

File: src/app/Scenes/Article/ArticleScreen.tsx [hunk 1]
```diff
@@ -16,30 +35,117 @@ export const ArticleScreen: React.FC<ArticleScreenProps> = (props) => {
...
 const Placeholder: React.FC = () => {
   return (
-    <ProvidePlaceholderContext>
-      <PlaceholderGrid />
-    </ProvidePlaceholderContext>
+    <Screen>
+      <Screen.Header rightElements={<ShareButton article={null as any} />} onBack={goBack} />
+
+      <Screen.Body fullwidth>
+        <Skeleton>
```
Comment (damassi) [inline src/app/Scenes/Article/ArticleScreen.tsx:94]: This is a more natural way of writing skeletons

File: src/app/Scenes/Article/ArticleScreen.tsx [hunk 2]
```diff
@@ -16,30 +35,117 @@ export const ArticleScreen: React.FC<ArticleScreenProps> = (props) => {
...
+          <Flex px={2}>
+            <Join separator={<Spacer y={0.5} />}>
+              <SkeletonText variant="xs" color="black100">
+                Art Vertical
+              </SkeletonText>
```
Comment (damassi) [inline src/app/Scenes/Article/ArticleScreen.tsx:99]: treat it like normal text, with variants, etc. Its how we do it on web

File: src/app/Scenes/Article/Components/Sections/ArticleSectionImageCollection/ArticleSectionImageCollectionCaption.tsx [hunk 3]
```diff
@@ -10,9 +11,26 @@ interface ArticleSectionImageCollectionCaptionProps {
...
+          tagStyles={{
+            p: {
+              color: color("black60"),
+              textAlign: "center",
+            },
+          }}
+        />
+      )}
+    </>
+  )
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/app/Scenes/Article/Components/Sections/ArticleSectionImageCollection/ArticleSectionImageCollectionCaption.tsx]: done

**Comment by damassi** [comment]
@gkartalis - this is ready for another review

File: src/app/Scenes/Article/Components/Sections/ArticleSectionImageCollection/ArticleSectionImageCollectionImage.tsx [hunk 4]
```diff
@@ -10,12 +13,33 @@ interface ArticleSectionImageCollectionImageProps {
...
+      <FastImage
+        style={dimensions}
+        onLoadStart={() => setLoading(true)}
+        onLoadEnd={() => setLoading(false)}
+        source={{
+          uri: data.image.resized.src,
+          priority: FastImage.priority.normal,
+        }}
+      />
+    </Flex>
```
Comment (damassi) [inline src/app/Scenes/Article/Components/Sections/ArticleSectionImageCollection/ArticleSectionImageCollectionImage.tsx:42]: Mentioned this in slack but thinking we could explore simplifying the opaque image view and pushing it over to Palette. Start with bare bones, and then go from there

File: src/app/Scenes/Article/ArticleScreen.tsx (hunk 1 above)
Comment (gkartalis) [reply src/app/Scenes/Article/ArticleScreen.tsx:98]: way more natural indeed 🔥🔥🔥 and you can quickly understand quickly which section is doing what! 

Screenshot 2023-06-15 at 14 53 57

File: src/app/Scenes/Article/Components/ArticleShareButton.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+      await RNShare.open({
+        title: data.title ?? "",
+        message: message + "\n" + url,
+        failOnCancel: false,
+      })
+    } catch (error) {
+      console.error("[ArticleScreen] Error sharing article:", error)
+    }
+  }
```
Comment (gkartalis) [inline src/app/Scenes/Article/Components/ArticleShareButton.tsx:28]: question(non blocking): do we want to also track analytics here for the sharing button? (can def happen on a follow up)

**Review by gkartalis — DISMISSED** [review]
Awesome work! 

While reviewing and reading through the code I was so excited to see the `<Screen>` component being used and also the `<Skeletons>` as well was really interesting!!! 

Thanks a lot 👏

File: src/app/Scenes/Article/Components/ArticleShareButton.tsx (hunk 5 above)
Comment (damassi) [reply src/app/Scenes/Article/Components/ArticleShareButton.tsx:28]: Ah yes! i was gonna ping abhiti today about this

File: src/app/Scenes/Article/ArticleScreen.tsx (hunk 1 above)
Comment (damassi) [reply src/app/Scenes/Article/ArticleScreen.tsx:98]: Yep exactly :) Looking at skeletons becomes real dumb and literal and 1:1... can pretty much just copy / paste code straight out of where yr using it and prepend `Skeleton` in the component name (`<Text variant='foo'>` -> `<SkeletonText variant='foo'>`. So, also quicker to build out.

File: src/app/Scenes/Article/Components/Sections/ArticleSectionImageCollection/__tests__/ArticleSectionImageCollectionCaption.tests.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+describe("ArticleSectionImageCollectionCaption", () => {
+  const Article = () => {
+    const data = useLazyLoadQuery<ArticleSectionImageCollectionCaptionTestQuery>(
+      graphql`
+        query ArticleSectionImageCollectionCaptionTestQuery {
+          article(id: "foo") {
+            sections {
+              ... on ArticleSectionImageCollection {
+                figures {
```
Comment (damassi) [inline src/app/Scenes/Article/Components/Sections/ArticleSectionImageCollection/__tests__/ArticleSectionImageCollectionCaption.tests.tsx:17]: Had to do it again here

File: src/app/Scenes/Article/Components/ArticleShareButton.tsx (hunk 5 above)
Comment (damassi) [reply src/app/Scenes/Article/Components/ArticleShareButton.tsx]: done in https://github.com/artsy/eigen/pull/8869/commits/45bc6d3f29b54d8e64e684fa35ea251c3d89d375

File: src/app/Scenes/Article/Components/Sections/ArticleSectionImageCollection/ArticleSectionImageCollectionImage.tsx (hunk 4 above)
Comment (damassi) [reply src/app/Scenes/Article/Components/Sections/ArticleSectionImageCollection/ArticleSectionImageCollectionImage.tsx]: done in https://github.com/artsy/palette-mobile/pull/117

**Pull request merged**